src/main/java/com/cloudbees/hudson/plugins/folder/computed/DefaultOrphanedItemStrategy.java

```diff
@@ -230,6 +255,43 @@ private static boolean disabled(TopLevelItem item) {
 
     @Override
     public <I extends TopLevelItem> Collection<I> orphanedItems(ComputedFolder<I> owner, Collection<I> orphaned, TaskListener listener) throws IOException, InterruptedException {
+        if (abortBuilds) {
+            Queue<Run<?, ?>> abortedBuilds = new LinkedList<>();
+            for (I item: orphaned) {
+                for (Job<?, ?> job: item.getAllJobs()) {
+                    if (job instanceof hudson.model.Queue.Task) {
+                        hudson.model.Queue jenkinsQueue = Jenkins.get().getQueue();
+                        hudson.model.Queue.Task task = (hudson.model.Queue.Task) job;
+                        for (hudson.model.Queue.Item pendingBuild: jenkinsQueue.getItems(task)) {
+                            jenkinsQueue.cancel(pendingBuild);
+                        }
+                    }
+                    for (Run<?, ?> build: job.getBuilds()) {
+                        Executor executor = build.getExecutor();
+                        if (executor == null) {
+                            // Since the latest builds are returned first and to avoid looping on a huge build history,
+                            // cut off the search when encountering a completed build.
+                            // Not totally correct, since it is possible for there to be a running build older than
+                            // the last completed build, but good enough as a heuristic.
+                            break;
+                        }
+                        executor.interrupt(Result.ABORTED, new OrphanedParent(item));
+                        abortedBuilds.add(build);
+                    }
+                }
+            }
+            long waitForAbortedBuildsStartTimeMillis = System.currentTimeMillis();
+            long maxWaitMillis = 60_000;
+            Run<?, ?> abortedBuild;
+            ABORTED_BUILDS: while ((abortedBuild = abortedBuilds.poll()) != null) {
+                while (abortedBuild.isLogUpdated()) {
+                    if ((System.currentTimeMillis() - waitForAbortedBuildsStartTimeMillis) > maxWaitMillis) {
+                        break ABORTED_BUILDS;
+                    }
+                    Thread.sleep(100L);
```

Comment:
By now I have already forgotten why we need to wait for the aborted builds to complete at all…so it would be a good idea to add a comment explaining the need for the benefit of future generations.